lean4-infoview/src/infoview/interactiveCode.tsx

```diff
@@ -153,17 +158,23 @@ function InteractiveCodeTag({tag: ct, fmt}: InteractiveTagProps<SubexprInfo>) {
     spanClassName += DIFF_TAG_TO_CLASS[ct.diffStatus] + ' '
   }
 
+  // ID that we can use to identify the component that a context menu was opened in.
+  // When selecting a custom context menu entry, VS Code will execute a VS Code command
+  // parameterized with `data-vscode-context`. We then use this context to execute the
+  // command in the context of the correct interactive code tag in the InfoView.
+  const interactiveCodeTagId = React.useId()
+  const vscodeContext = { interactiveCodeTagId }
+  useEvent(ec.events.goToDefinition, _ => execGoToLoc(), undefined, interactiveCodeTagId)
+
   return (
     <WithTooltipOnHover
+      data-vscode-context={JSON.stringify(vscodeContext)}
```

Comment:
I am wondering how difficult it would be to store enough information here for the editor extension to be able to go-to-def without sending a message back to the infoview. The main issue appears to be that we don't eagerly know the location one has to go to (`fetchGoToLoc` does that). We could in principle only set this context after `fetchGoToLoc` has replied, but that might make for some bad UX wherein the 'Go to Definition' option is sometimes there and sometimes not (or is always there, but doesn't always work).

Reply:
Yes, I didn't see a nice way to implement this differently, either. 

We could in principle invert the control flow: The 'Go to definition' option is always displayed but when clicked waits until the infoview fires an event for the given id either with a location or undefined. I don't think this is much better, though.

Reply:
There are no good solutions here 😅